cvat-sdk/cvat_sdk/pytorch/project_dataset.py

```diff
@@ -42,6 +42,7 @@ def __init__(
         label_name_to_index: Mapping[str, int] = None,
         task_filter: Optional[Callable[[models.ITaskRead], bool]] = None,
         include_subsets: Optional[Container[str]] = None,
+        update_policy: UpdatePolicy = UpdatePolicy.IF_MISSING_OR_STALE,
```

Comment:
I suggest to allow `CacheManager` instances too, especially in the `TaskVisionDataset` class. This should simplify testing, using custom cache controllers, and, the most important, will allow to use a single cache controller in the multi-threaded scenario.

Reply:
I don't want to expose the `CacheManager` interface to the public; it's pretty ad-hoc, and I didn't design it to be user-extensible.

> and, the most important, will allow to use a single cache controller in the multi-threaded scenario.

Can you elaborate on this?